src/mesh/BoundingBox.hpp

```diff
@@ -45,9 +53,14 @@ class BoundingBox {
   /// Comparison Operator
   bool operator==(const BoundingBox &otherBB) const;
 
-  /// Check whether the bounding box is at default state or not
+  /// empty() returning true means that the length in every BoundingBox's dimension is equal to zero
   bool empty() const;
 
+  /// Check whether the bounding box is at default state or not
+  /// all the values of _boundMin = std::numeric_limits<double>::max()
+  /// all the values of _boundMax = std::numeric_limits<double>::lowest()
+  bool isDefault() const;
```

Comment:
Maybe it would be useful to document (here or at least in the pull request conversation) what the purpose of the `empty` and of the `isDefault` is. At which situation is a box expected to be empty and in which default?